Gateway/Command/MidpageCancelCommand.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Billink\Billink\Gateway\Command;
+
+use Billink\Billink\Model\Payment\MidpageCancelService;
+use Billink\Billink\Model\Payment\OrderHistory;
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Payment\Gateway\CommandInterface;
+use Magento\Payment\Gateway\Helper\ContextHelper;
+use Magento\Payment\Gateway\Helper\SubjectReader;
+use Magento\Sales\Model\Order\Payment;
+use Psr\Log\LoggerInterface;
+
+class MidpageCancelCommand implements CommandInterface
+{
+    private CheckoutSession $session;
+    private LoggerInterface $logger;
+    private OrderHistory $orderHistory;
+    private MidpageCancelService $cancelService;
+
+    public function __construct(
+        CheckoutSession $session,
+        LoggerInterface $logger,
+        OrderHistory $orderHistory,
+        MidpageCancelService $cancelService
+    ) {
+        $this->session = $session;
+        $this->logger = $logger;
+        $this->orderHistory = $orderHistory;
+        $this->cancelService = $cancelService;
+    }
+
+    /**
+     * @param array $commandSubject
+     * @return void
+     * @throws LocalizedException
+     */
+    public function execute(array $commandSubject): void
+    {
+        $paymentDO = SubjectReader::readPayment($commandSubject);
+        /** @var Payment $payment */
+        $payment = $paymentDO->getPayment();
+        ContextHelper::assertOrderPayment($payment);
```

Comment:
can this assert throw? if so, should we catch -> log -> throw again?

Reply:
Yes, but not sure if we need to log it. If something is broken at this point - it's more for the magento/development flow, than to "something with the data". So this assert makes sure that correct object is provided and that the main class that builds it is not called from some other place with some other params.